Kompleksowa aktualizacja: Nowy downloader, odświeżony codebase Android/iOS oraz dokumentacja

n1ghtmare13 opened the pull request:
### Zmiany:

- **Nowe narzędzie:** Skonsolidowano funkcjonalność trzech dotychczasowych plików (`clean_line_numbers.py`, `organize_files.py` oraz `mobywatel-downloader.user.js`) w jeden nowoczesny `mobywatel-downloader-userscript.js`. 
  - Skrypt umożliwia natychmiastowe generowanie i pobieranie całego codebase w formie nieskompresowanego archiwum ZIP bezpośrednio z przeglądarki.
  - Wyeliminowano wymóg posiadania środowiska Python (runtime).
  - Kod narzędzia został zoptymalizowany i skrócony o kilkaset linii przy zachowaniu pełnej logiki czyszczenia kodu z artefaktów (numery linii).
- **Odświeżony codebase (Android/iOS):** W ramach tego PR zastąpiono obecne zasoby w katalogach `Android/` i `iOS/` nowym zrzutem wykonanym za pomocą powyższego narzędzia. Pozwoliło to na:
  1. Naprawienie błędów zgłoszonych w #4 ( Poprawę błędnych rozszerzeń plików (np. zamiana `.iml.html` na poprawny `.iml`).
  2. Rozwiązanie problemów z brakującymi plikami w codebase iOS oraz błędami przy pobieraniu plików binarnych (uszkodzone obrazy), które występowały w poprzedniej wersji skryptu.
- **Dokumentacja:** Kompletne odświeżenie `README.md` oraz jego wersji angielskiej `README-en.md`. Wprowadzono kluczowe sekcje dotyczące bezpieczeństwa (aspekty prawne i techniczne) oraz zaktualizowano instrukcję instalacji.
- **Licencja:** Przebudowano plik `LICENSE`, aby jednoznacznie rozgraniczyć licencję MIT dla narzędzi od licencji kodu źródłowego COI, uwzględniając poprawne noty o prawach autorskich. Dodatkowo pliki licencji MIT od COI zostały dodane bezpośrednio do podkatalogów z kodem (również przez skrypt).

### Powiązania:
- **Zamyka #4 :** Przy okazji użycia nowej wersji skryptu zostały naprawione błędne rozszerzenia.
- PR ten zawiera i finalizuje zmiany zaproponowane w moim poprzednim Pull Request dotyczącym samych narzędzi (który w tym momencie zamykam).

n1ghtmare13 committed:
chore: usunięcie starych skryptów Python i poprzedniej wersji downloadera

n1ghtmare13 committed:
feat: dodanie zintegrowanego skryptu userscript z auto-czyszczeniem i ZIP

n1ghtmare13 committed:
chore: aktualizacja licencji o autorstwo narzędzi i zastrzeżenia dot. COI

n1ghtmare13 committed:
docs: aktualizacja instrukcji obsługi oraz dodanie sekcji o bezpieczeństwie

n1ghtmare13 committed:
feat: odświeżenie codebase Android/iOS za pomocą nowej wersji skryptu

- Wykorzystano ulepszoną wersję narzędzia, która automatycznie poprawia rozszerzenia plików (usunięcie błędnych końcówek .html)
- Przywrócono brakujące pliki źródłowe oraz zasoby graficzne, których nie pobierał stary skrypt
- Zastąpiono wadliwe/uszkodzone zasoby nowymi, zapewniając pełną zgodność ze strukturą oryginalnego repozytorium

fajfer commented:
@n1ghtmare13 tu akurat _mea culpa_ za nie umieszczenie licencji na plikach stworzonych przeze mnie, ale czy byłoby to dla Ciebie OK żebyśmy ustalili licencję jako [EUPL 1.2](https://eupl.eu/1.2/pl/) zamiast MIT? Argumentuję tym, że to najpopularniejsza wolna licencja oprogramowania w języku polskim (np. MIT umieszczone przez COI jest jedynie tłumaczeniem oryginalnej licencji) i została stworzona z myślą o zastosowaniu w administracji publicznej. Wygląda jak idealny przypadek użycia.

Jeżeli to dla Ciebie ok to taką ustawiłbym EUPL 1.2 na repozytorium a pliki mObywatela już zgodnie z przetłumaczoną MIT

0x8008 commented:
Dyskusja i PR-y na wyższym poziomie niż w 99% repów OSS

0x8008 commented:
> > Dyskusja i PR-y na wyższym poziomie niż w 99% repów OSS
> 
> Teraz obniżyłem poziom wzmianką o toalecie

<img width="500" height="379" alt="image" src="https://github.com/user-attachments/assets/4ea6be6f-54fa-486a-b373-616f79ab5864" />

n1ghtmare13 committed:
fix: Poprawka formatowania author oraz lepszy opis w metadata header

n1ghtmare13 committed:
fix: Zmiana rozszerzenia skryptu scrapującego na zgodną ze standardem userscriptów

Nazwa zmieniona z:
mobywatel-downloader-userscript.js
na:
mobywatel-downloader.user.js

n1ghtmare13 committed:
docs: ujednolicenie README i nazwy pliku userscript

fajfer commented:
@n1ghtmare13 odkładam to tak w czasie, bo nie wiem jak ugryźć temat ładowania modułów na firefoxie. Problem jest taki, że to co zrobiłeś jest spoko i zadziała na bazujących na chromium ale już nie na FF. Nie wiem czy teraz wymyślać alternatywny sposób na jakiś bundling tych plików niż zipowanie (tak, aby uniknąć komplikacji i dorzucania dodatkowych zależności).

Dorzucanie jakiegoś kodu z fflate do tego już sprawiłoby, że cały skrypt stałby się mniej czytelny. Daj znać co myślisz, ja też spróbuję pomyśleć ale trochę mam rzeczy na głowie.

n1ghtmare13 commented:
@fajfer możesz konkretniej wskazać problem (może być nawet issue w moim branchu), bo testowałem swój skrypt na 3 przeglądarkach i dynamiczny import nie sprawiał żadnego problemu? (sam na co dzień korzystam z przeglądarki firefox-based).
Czytałeś zaktualizowane README? Może używasz innego script managera. Zarekomendowałem ViolentMonkey który jest maintained w przeciwieństwie do Greasemonkey, który jest chyba martwy i nikt go nie zaleca.

n1ghtmare13 commented:
Jeśli to jakiś wymóg, mogę zmienic sposób pobierania biblioteki na klasyczne require z umd ale nie sądzę aby dynamiczny import tak lekkiej biblioteki miał sprawiać jakiś problem.
